### PR TITLE
Moved all VecDeque tests from libcollections to libcollectionstest

### PR DESCRIPTION
**Pull request by KiChjang**
Encountered while I was trying to work on #28518. Please let me know whether this is a correct move.

### CODE REVIEW

**Comment by bluss** [comment]
I found the travis error, search the log for "duplicate definition of value \`test_drain\`"

**Comment by KiChjang** [comment]
Oh, which test should we use for `test_drain` in that case? There are two tests and I don't know which of them is superior.

**Comment by bluss** [comment]
Save both. One is testing `drain(..)` and the other is testing with different ranges.

**Comment by bluss** [comment]
Why do we move these? I don't really know why we have two crates.

**Comment by KiChjang** [comment]
@Gankro said this on IRC:
```
<Gankro> KiChjang: used to be everything was in libcollections, but libcollectionstest has to pull some stuff out when libcollections gained some langitemss
```

**Comment by bluss** [comment]
Ok good

@bors r+

**Comment by KiChjang** [comment]
```
test num::tests::test_i32f64 ... ok

test num::tests::test_i32i64 ... ok

<test>:1:62: 1:63 error: macro undefined: 'z!'

<test>:1 fn bogus() {macro_rules! z (() => (3+4));}fn inty() -> i32 { z!() }

                                                                      ^

<test>:1:62: 1:63 help: did you mean `z!`?

error: aborting due to previous error

test ext::expand::tests::macros_cant_escape_fns_test ... ok

test ext::expand::tests::method_arg_hygiene ... ok

test num::tests::test_i8f64 ... ok

test num::tests::test_i8f32 ... ok

test ext::expand::tests::method_arg_hygiene_2 ... ok

<test>:1:59: 1:60 error: macro undefined: 'z!'

<test>:1 mod foo {macro_rules! z (() => (3+4));}fn inty() -> i32 { z!() }

                                                                   ^

<test>:1:59: 1:60 help: did you mean `z!`?

error: aborting due to previous error

test ext::expand::tests::macros_cant_escape_mods_test ... ok
```
Did I do anything wrong that caused this output?

File: src/test/run-pass/zero-sized-vec-deque-push.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+use std::collections::VecDeque;
+use std::iter::Iterator;
+
+#[test]
+fn test_zst_push() {
```
Comment (bluss) [inline src/test/run-pass/zero-sized-vec-deque-push.rs]: In a run-pass test, you need a main function and just call this test function manually. We don't use `#[test]` here.

**Comment by bluss** [comment]
@bors r+

Thanks! Notice that there is no notification for new pushes to a PR. You need to put a message here to notify us who are subscribing.

**Pull request merged**